site/docs/hosting/vps.md

```diff
@@ -2,9 +2,84 @@
 
 A virtual private server, mostly known as VPS, is a virtual machine running in the cloud with its users having the full control of its system.
 
-In this guide, you'll learn about various methods of running your bot in a VPS, keeping it online 24/7, making it run automatically when your VPS boots and restart on crashes.
+## Preparing Your Code
 
-## systemd
+To get started, you can use the bot shown in the [example of using a session](../plugins/session.md#example-usage).
+
+In the following, we will assume that the bot runs on Node.js
+Why only Node.js and not Deno or is it better for both runtimes?
+The answer is simple: the Deno pipeline would be very simple.
+A Node.js pipeline is complicated by the additional task of building and setting up communication between the two tasks.
+So a CI/CD pipeline for Node.js would be a better example.
+
+## Server Rental
+
+We chose [Hostinger](https://hostinger.com/) as our host.
+There are several reasons for this:
+
+1. It is multilingual, so you will find it convenient and easy to use.
+2. It has a very rich documentation that thoroughly covers all the problem areas that a beginner may encounter.
+3. It offers powerful servers at fairly low prices.
+
+:::tip Analog of a server
+If you cannot or do not want to rent a server, you can replace it with a virtual machine.
+To do this, use an application such as [VirtualBox] (https://virtualbox.org/).
+Create a virtual machine with the desired Linux distribution and use it as a real server.
+:::
+
+Go to the [VPS tariff catalog](https://hostinger.com/vps-hosting/).
+We will use the "VPS 1" plan.
+Below you can see a comparison table that describes the offered server resources and limits.
+The resources of VPS 1 are enough for bots with a small audience, and even more so for our test bot.
+
+Go back to the "VPS 1" card and click the "Add" button.
+You will be automatically redirected to the checkout page, where you will also immediately register with Hostinger.
+
+:::warning Change the rental term!
+The typical lease term is 2 years (a marketing ploy), and it costs a lot of money.
+You probably don't need it, so you can start by renting a server for a month, which is much cheaper.
+
+In any case, Hostinger provides a 30-day money-back guarantee.
+:::
+
+After payment you will be able to set up your server:
+
+1. **Location.**
+2. **Server type.**
+   Choose the option "Clean OS".
+3. **Operating system.**
+   We will use Ubuntu 22.04.
+   If you choose a different system, some steps may be different, so be careful.
+4. **Server name.**
+5. **Root password.**
+   Make a strong password and keep it in a safe place!
+6. **SSH key**.
+   Skip this step.
+
+After the server is created, you can connect to it using SSH:
+
+> SSH (_Secure Shell_) is a network protocol for remote control of a computer (server) and file transfer.
+
+```sh:no-line-numbers
+ssh root@<ip-address>
```

Comment:
Should we tell them not to work with the root user?

Reply:
We can leave a note about it. The problem with denying root user is the subsequent configuration of allowing `sudo systemctl restart <service-name>` without having to enter a password during CI/CD. This adds unnecessary complexity to the article, and there is little benefit, because it is unlikely that the user of the article will write corporate bots for which it will be necessary to think about security (this is just my opinion). Otherwise, a separate professional will be responsible for deploying the bot, or the bot developer will study security issues on his own.

Reply:
Fair point. Lets add a note that this would be good to do, but it extends the scope of the article, so people should research it on their own